pkg/querier/store.go

```diff
@@ -92,9 +113,99 @@ outer:
 			}
 			iterators = append(iterators, iterator)
 		}
+
 		result = append(result, iter.NewNonOverlappingIterator(iterators, labels))
 	}
-	return result, nil
+
+	return iter.NewHeapIterator(result, req.Direction), nil
+}
+
+func loadFirstChunks(ctx context.Context, chks map[model.Fingerprint][][]chunkenc.LazyChunk) error {
+	sp, ctx := opentracing.StartSpanFromContext(ctx, "loadFirstChunks")
+	defer sp.Finish()
+
+	// If chunks span buckets, then we'll have different fetchers for each bucket.
+	chksByFetcher := map[*chunk.Fetcher][]*chunkenc.LazyChunk{}
+	for _, lchks := range chks {
+		for _, lchk := range lchks {
+			chksByFetcher[lchk[0].Fetcher] = append(chksByFetcher[lchk[0].Fetcher], &lchk[0])
```

Comment:
not sure if possible, but you could have a index out of boundaries.

Reply:
Fixed it with a check above.